www/qserv/js/ToolsSql.js

```diff
@@ -0,0 +1,219 @@
+                this._password().val('');
+                this._query().val('');
+            });
+        }
+
+        // Inputs
+
+        _worker() {
+            if (this._worker_obj === undefined) {
+                this._worker_obj = this.fwk_app_container.find('input#worker');
+            }
+            return this._worker_obj;
+        }
+        _user() {
+            if (this._user_obj === undefined) {
+                this._user_obj = this.fwk_app_container.find('input#user');
+            }
+            return this._user_obj;
+        }
+        _password() {
+            if (this._password_obj === undefined) {
+                this._password_obj = this.fwk_app_container.find('input#password');
+            }
+            return this._password_obj;
+        }
+        _query() {
+            if (this._query_obj === undefined) {
+                this._query_obj = this.fwk_app_container.find('textarea#query');
+            }
+            return this._query_obj;
+        }
+
+        /**
+         * The table for displaying result sets of queries
+         * @returns JQuery table object
+         */
+        _table() {
+            if (this._table_obj === undefined) {
+                this._table_obj = this.fwk_app_container.find('table#fwk-tools-sql-resultset');
+            }
+            return this._table_obj;
+        }
+
+        /**
+         * Load data from a web servie then render it to the application's page.
+         */
+        _load() {
+            if (this._loading === undefined) {
+                this._loading = false;
+            }
+            if (this._loading) return;
+            this._loading = true;
+
+            if (this._table().children('caption').html() === '') this._table().children('caption').html('updating...');
+            this._table().children('caption').removeClass('error').addClass('updating');
+            Fwk.web_service_POST(
+                "/replication/v1/sql/query",
+                {   'worker':   this._worker().val(),
+                    'user':     this._user().val(),
+                    'password': this._password().val(),
+                    'query':    this._query().val()
+                },
+                (data) => {
+                    Fwk.setLastUpdate(this._table().children('caption'));
+                    this._table().children('caption').removeClass('updating');
+                    this._display(data);
+                    this._loading = false;
+                },
+                (msg) => {
+                    this._table().children('caption').removeClass('updating');
+                    this._table().children('caption').addClass('error').text(msg);
+                    this._loading = false;
+                }
+            );
+        }
+        
+        /**
+         * Analyze and render result sets or error
+         */
+        _display(data) {
+            console.log(data);
+            if (data.status !== 'FINISHED::SUCCESS::EXT_STATUS_NONE') {
```

Comment:
Magic!

Reply:
Good point. I could add a boolean flag to indicate if a result of an operation was successful.